src/main/java/net/sf/jabref/logic/importer/MrDLibFetcher.java

```diff
@@ -0,0 +1,318 @@
+                boolean published_in;
+                boolean title;
+                boolean year;
+                BibEntry currentEntry;
+
+
+                @Override
+                public void startElement(String uri, String localName, String qName, Attributes attributes)
+                        throws SAXException {
+
+                    if (qName.equalsIgnoreCase("related_article")) {
+                        currentEntry = new BibEntry();
+                    }
+                    if (qName.equalsIgnoreCase("authors")) {
+                        authors = true;
+                    }
+                    if (qName.equalsIgnoreCase("published_in")) {
+                        published_in = true;
+                    }
+                    if (qName.equalsIgnoreCase("title")) {
+                        title = true;
+                    }
+                    if (qName.equalsIgnoreCase("year")) {
+                        year = true;
+                    }
+
+                }
+
+                @Override
+                public void endElement(String uri, String localName, String qName) throws SAXException {
+                    if (qName.equalsIgnoreCase("related_article")) {
+                        bibEntryList.add(currentEntry);
+                        currentEntry = new BibEntry();
+                    }
+                }
+
+                @Override
+                public void characters(char ch[], int start, int length) throws SAXException {
+
+                    if (authors) {
+                        currentEntry.setField("author", new String(ch, start, length));
+                        authors = false;
+                    }
+                    if (published_in) {
+                        currentEntry.setField("journal", new String(ch, start, length));
+                        published_in = false;
+                    }
+                    if (title) {
+                        currentEntry.setField("title", new String(ch, start, length));
+                        title = false;
+                    }
+                    if (year) {
+                        currentEntry.setField("year", new String(ch, start, length));
+                        year = false;
+                    }
+
+
+                }
+
+            };
+
+            try {
+                InputStream stream = new ByteArrayInputStream(recomendations.getBytes());
+                saxParser.parse(stream, handler);
+            } catch (Exception e) {
+                LOGGER.error(e.getMessage(), e);
+            }
+        } catch (ParserConfigurationException e) {
+            LOGGER.error(e.getMessage(), e);
+        } catch (SAXException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+        return bibEntryList;
+    }
+
+    /**
+     * Converts the xmlDocument to htmlSnippets using a SaxParser
+     * @param xmlDocument
+     * @return
+     */
+    private List<String> convertToHtml(String recommendations) {
```

Comment:
It will be not possible to use this method again for another setting. It is very tight coupled to MrDlib, rising from the document we deliver. Should it move though?